src/main/java/Duke.java

```diff
@@ -1,10 +1,124 @@
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Scanner;
+
 public class Duke {
+
+    private static final String DIVIDER = "    ____________________________________________________________";
+    private static List<Task> store = new ArrayList<Task>();
+
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        Scanner scan = new Scanner(System.in);
+        boolean exit = false;
+        greet();
+
+        while (true) {
+            String command = scan.nextLine();
+            HashMap<String, String> commandArgs = parseCommand(command);
+            System.out.println(DIVIDER);
+
+            switch (commandArgs.get("command")) {
+            case "list":
+                list();
+                break;
+            case "done":
+                setTaskDone(Integer.parseInt(commandArgs.get("param")));
+                break;
+            case "todo":
+                addTodo(commandArgs.get("param"));
+                break;
+            case "deadline":
+                addDeadline(commandArgs.get("param"), commandArgs.get("by"));
+                break;
+            case "event":
+                addEvent(commandArgs.get("param"), commandArgs.get("at"));
+                break;
+            case "bye":
+                exit = true;
+                bye();
+                break;
+            default:
+                break;
+            }
+
+            System.out.println(DIVIDER);
+
+            if (exit) {
+                break;
+            } else {
+                System.out.println("");
+            }
+        }
+
+        scan.close();
+    }
+
+    public static HashMap<String, String> parseCommand(String command) {
+        String[] parts = command.split(" /");
+        HashMap<String, String> map = new HashMap<String, String>();
+        String[] commandParts = parts[0].split(" ", 2);
+        map.put("command", commandParts[0]);
+        map.put("param", commandParts.length > 1 ? commandParts[1] : "");
+        for (int i = 1; i < parts.length; i++) {
+            String[] argParts = parts[i].split(" ", 2);
+            map.put(argParts[0], argParts[1]);
+        }
+        return map;
+    }
+
+    public static void greet() {
+        System.out.println(DIVIDER);
+        System.out.println("     Hello! I'm Duke");
```

Comment:
Should this space characters be extracted out to a constant? It will make the code look cleaner and make it easier to adjust the spaces should it change

Reply:
Yea constant

Reply:
Yes, definitely!